### PR TITLE
Add region yolo reference implementation

### PR DESCRIPTION
**Pull request by YangleiZouIntel**
This PR adds region yolo reference implementation from upstream 2021.2

### CODE REVIEW

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/region_yolo.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    auto param = std::make_shared<ngraph::op::Parameter>(ngraph::element::f32, inputShape);
+    auto region_yolo = std::make_shared<ngraph::op::v0::RegionYolo>(param, coords, classes, num_regions, do_softmax, mask, start_axis, end_axis);
+    function = std::make_shared<ngraph::Function>(std::make_shared<ngraph::opset1::Result>(region_yolo), ngraph::ParameterVector{param}, "RegionYolo");
+}
+
+TEST_P(RegionYoloLayerTest, CompareWithRefs) {
+    Run();
+};
+
+} // namespace LayerTestsDefinitions
```
Comment (tzerrell) [inline inference-engine/tests/functional/plugin/shared/src/single_layer_tests/region_yolo.cpp]: Add a newline character at the end of the file.

File: ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                                           width);
+                    }
+                }
+            }
+
+        } // namespace reference
+
+    } // namespace runtime
+
+} // namespace ngraph
```
Comment (tzerrell) [inline ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp]: Add newline

**Review by tzerrell — COMMENTED** [review]
LGTM; 2 linting comments

File: ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp (hunk 2 above)
Comment (YangleiZouIntel) [reply ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp]: Done

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/region_yolo.cpp (hunk 1 above)
Comment (YangleiZouIntel) [reply inference-engine/tests/functional/plugin/shared/src/single_layer_tests/region_yolo.cpp]: Done

**Review by tzerrell — APPROVED** [review]

**Pull request merged**